File: core/src/com/alta_v2/rendering/menuScreen/MenuScreen.java
package com.alta_v2.rendering.menuScreen;

import com.alta_v2.game.utils.Resources;
import com.alta_v2.rendering.Renderer;
import com.alta_v2.rendering.ScreenState;
import com.alta_v2.utils.AssetLoader;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.google.inject.assistedinject.AssistedInject;
import lombok.extern.log4j.Log4j2;

/**
 * Provides the menu screen.
 */
@Log4j2
public class MenuScreen implements Renderer {

    private final AssetManager assetManager;

    private Batch batch;
    private Texture backgroundTexture;


    /**
     * Initialize new instance of {@link MenuScreen}.
     */
    @AssistedInject
    public MenuScreen() {
        this.assetManager = AssetLoader.load(Resources.TEXTURE_MENU_BACKGROUND);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void init(ScreenState state) {
        this.backgroundTexture = this.assetManager.get(Resources.TEXTURE_MENU_BACKGROUND, Texture.class);
        this.batch = new SpriteBatch();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void render(float delta, ScreenState state) {
        this.batch.begin();
        this.batch.draw(this.backgroundTexture, 0, 0, Gdx.graphics.getWidth(), Gdx.graphics.getHeight());
        this.batch.end();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void dispose() {
        this.batch.dispose();
        this.backgroundTexture.dispose();
        this.assetManager.dispose();
    }
}

File: core/src/com/alta_v2/rendering/menuScreen/MenuScreen.kt
package com.alta_v2.rendering.menuScreen

import com.alta_v2.game.utils.Resources
import com.alta_v2.rendering.Renderer
import com.alta_v2.rendering.ScreenState
import com.alta_v2.utils.AssetLoader.Companion.load
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.assets.AssetManager
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.google.inject.assistedinject.AssistedInject

/**
 * Provides the menu screen.
 */
class MenuScreen @AssistedInject constructor() : Renderer {

    private lateinit var assetManager: AssetManager
    private lateinit var batch: Batch
    private lateinit var backgroundTexture: Texture

    /**
     * {@inheritDoc}
     */
    override fun init(state: ScreenState) {
        assetManager = load(Resources.TEXTURE_MENU_BACKGROUND)
        backgroundTexture = assetManager.get(Resources.TEXTURE_MENU_BACKGROUND, Texture::class.java)
        batch = SpriteBatch()
    }

    /**
     * {@inheritDoc}
     */
    override fun render(delta: Float, state: ScreenState) {
        batch.begin()
        batch.draw(backgroundTexture, 0f, 0f, Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat())
        batch.end()
    }

    /**
     * {@inheritDoc}
     */
    override fun destroy() {
        batch.dispose()
        backgroundTexture.dispose()
        assetManager.dispose()
    }
}